amqp/message_sender.go

```diff
@@ -85,12 +78,3 @@ func failOnError(err error, msg string) {
 		log.Fatalf("%s: %s", msg, err)
 	}
 }
-
```

Comment:
I like the hasError method but it had a possible bug, because we are using it on a routine. When we change the return and body of response maybe we already sent the result to the user.